state/core_access.go

```diff
@@ -104,20 +111,57 @@ func NewCoreAccessor(
 		defaultSignerAddress: addr,
 		getter:               getter,
 		prt:                  prt,
-		coreConn:             conn,
+		coreConns:            conns,
 		network:              network,
 	}
+
+	// Initialize the nextConnIndex to 0
+	ca.nextConnIndex.Store(0)
+
 	return ca, nil
 }
 
+// getNextConn returns the next connection in a round-robin fashion and advances
+// the internal counter. This provides simple load balancing across
+// multiple Core gRPC endpoints.
+func (ca *CoreAccessor) getNextConn() *grpc.ClientConn {
+	// If we only have one connection, just return it
+	if len(ca.coreConns) <= 1 {
+		return ca.coreConns[0]
```

Comment:
`len <=1` means `0` or `1` on this line. For `0` this line will panic. Is this okay in this place?

Reply:
Looks like it's at least 1 by looking at `NewCoreAccessor`. So I suggest to change `<=` to `==`.

Reply:
Thank you 👍 Fixed it